### ISSUE TITLE
Embedded CSS in HTML headers and footer is ignored

### ISSUE THREAD

**Issue by olivierr91** [jsreport/jsreport-chrome-pdf#15]
I am trying to switch from PhantomPDF to Chrome in .NET and having some issues. I used to pass HTML / embedded CSS to headers and footers to Phantom and it worked perfectly. In Chrome embedded CSS is ignored. The documentation is not clear about what is exactly supported as header/footer content.

Here is my HTML header (this is Razor syntax, but you will understand what I want to achieve here):

```
<html>
<head>
    @Html.EmbedCss("css/documents/page-headers/report-page-header.css", hostingEnvironment)
</head>
<body>
    <div class="page-header">
        <img class="logo" src="data:image/png;base64, @Html.EmbedBase64("img/company-logo.png", hostingEnvironment)" />
        <table class="details">
            <tr>
                <td>@Localizer["GeneratedOn"]:&nbsp;</td>
                <td>@Model.GeneratedOn.ToString("G")</td>
            </tr>
            <tr>
                <td>@Localizer["GeneratedBy"]:&nbsp;</td>
                <td>@Model.GeneratedBy?.ToDisplayString() (@Model.GeneratedBy?.Username)</td>
            </tr>
        </table>
        <div class="title">
            @Model.Name
        </div>
    </div>
</body>
</html>
```

Here the PhantomPDF rendering result:

[commandes_dachat_avec_items_2018_07_18_10_10_08.pdf](https://github.com/jsreport/jsreport-chrome-pdf/files/2206110/commandes_dachat_avec_items_2018_07_18_10_10_08.pdf)

Here the Chrome rendering result with the same code:

[commandes_dachat_avec_items_2018_07_18_09_43_03.pdf](https://github.com/jsreport/jsreport-chrome-pdf/files/2206111/commandes_dachat_avec_items_2018_07_18_09_43_03.pdf)

**Comment by pofider** [comment]
The ` @Html.EmbedCss` creates `style` tag like this?

```html
  <style>
        .page-header {
            color: red; 
        }
    </style>
```

This makes the text in header red for me. 
As we mention in the docs. The headers in chrome/puppeteer are buggy.
https://jsreport.net/learn/chrome-pdf#native-headers-and-footers

In this thread I see some styles are not supported in header
https://stackoverflow.com/questions/49943479/puppeteer-header-and-footertemplate-doesnt-work
Not all styling properties are supported, you should avoid using position, top, bottom.

Here is also one thread mentioning it
https://github.com/GoogleChrome/puppeteer/issues/2388

**Comment by olivierr91** [comment]
Yes color works, but other basic HTML styles do not, such as body font-size, float, etc:

```
<html>
<head>
    <style>
        body {
            font-family: "Arial";
            font-size: 7px;
            font-weight: normal;
        }

        .page-header {
            color: blue;
        }

        .left {
            float: left;
        }

        .right {
            float: right;
        }
    </style>
</head>
<body>
    <div class="page-header">
        <div class="left">Left</div>
        <div class="right">Right</div>
    </div>
</body>
</html>
```

Chrome rendering is useless if we can't put basic stuff in the header. Basic as in more complex than a single line of text but enough to have things like a logo at the left, a title and some info on the right. Unless there is a workaround, I will have to postpone my switch to headless Chrome until Pupeteer fix their stuff.

**Comment by WalterWeidner** [comment]
I ran into similar issues with strange CSS quirks in the `headerTemplate` today. My code was using flexbox to position content at the bottom of my page's header. Positioning this way enables us to adjust the top margin of the page while maintaining header content in contact with page body. For some reason `pdf()` was positioning my header template under the page body's content (even though it works in a normal browser window).

We're going to have to postpone migration to Google Chrome until CSS header and footer support becomes more stable and hopefully support for external resources is added (https://github.com/GoogleChrome/puppeteer/issues/1853).

**Comment by abenhamdine** [comment]
FWIW, we also have to stop migration from phantom to Chrome because of background colors in table cells which are not displayed anymore with Chrome.

**Comment by MrMilan** [comment]
@abenhamdine 
FYI background-colors solve 
@media print { body { -webkit-print-color-adjust: exact; } }

https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-print-color-adjust

**Comment by MrMilan** [comment]
@abenhamdine 
I had same problem. However I find that is a problem of printing. "Intelligent" google thinking, too much. 
I solved with
`@media print { body { -webkit-print-color-adjust: exact; } }`

https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-print-color-adjust
